docs/rules.rst

```diff
@@ -0,0 +1,55 @@
+List of warnings
+------------------
+
+- **ASYNC100**: A ``with [trio/anyio].fail_after(...):`` or ``with [trio/anyio].move_on_after(...):`` context does not contain any ``await`` statements.  This makes it pointless, as the timeout can only be triggered by a checkpoint. This check also allows ``yield`` statements, since checkpoints can happen in the caller we yield to.
```

Comment:
> Let's add a "General rules" subheading for the 1xx rules.

:+1:

> I'd also consider using a sub-subheading for each individual rule (instead of the bullet points), so that it's easier to add code examples etc. That's probably better as a follow-up PR though.

It's either that or a separate subpage for each one (ruff style) as you suggested in https://github.com/python-trio/flake8-async/issues/214#issuecomment-1981529208

I'm not sure which one I prefer, tons of small pages can be messy to navigate - so maybe a dense list for an overview that then links to longer sections for each rule. A raw TOC wouldn't be great with error codes, but we probably should come up with human-friendly names for the codes at some point (https://github.com/astral-sh/ruff/issues/1773)

But yeah, leaving that one for followup as I said in https://github.com/python-trio/flake8-async/pull/223#issuecomment-2056378770